src/hamiltonian.jl

```diff
@@ -6,7 +6,11 @@ struct Hamiltonian{T<:Real,M<:AbstractMetric{T},F1,F2}
     ∂logπ∂θ     ::  F2
 end
 
-function ∂H∂θ(h::Hamiltonian, θ::AbstractVector{T}) where {T<:Real}
+function (h::Hamiltonian)(M⁻¹)
+    return Hamiltonian(h.metric(M⁻¹), h.logπ, h.∂logπ∂θ)
+end
```

Comment:
A bunch of callable type definitions like this is for conveniently making new copy of the same thing with updated parameters. E.g. here is to update the `metric` of `Hamiltonian`.

Reply:
I see - perhaps add a comment in the code.

Reply:
Added in a0af7ccc668a2a1e537dc5acdbe8c7bd22f11a86.